server/service/frontend.go

```diff
@@ -27,6 +27,14 @@ func ServeFrontend(urlPrefix string, sandbox bool, logger log.Logger) http.Handl
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		writeBrowserSecurityHeaders(w)
 
+		// The following check is to prevent a misconfigured osquery from submitting
+		// data to the root endpoint (the osquery remote API uses POST for all its endpoints).
+		// See https://github.com/fleetdm/fleet/issues/16182.
+		if r.Method == "POST" {
```

Comment:
@lucasmrod I just want to double check what happens when osquery will get this 405.
Will it retry? or do we know that getting 405 will make it drop the POST?

Reply:
Yes, it will retry.
See https://github.com/fleetdm/fleet/pull/16750#issuecomment-1941874594.